fastlane_bot/helpers/txhelpers.py

```diff
@@ -150,22 +153,32 @@ def check_and_approve_tokens(self, tokens: List):
             allowance = token_contract.caller.allowance(self.wallet_address, self.arb_contract.address)
             self.cfg.logger.info(f"Remaining allowance for token {token_address} = {allowance}")
             if allowance == 0:
-                tx = self._create_transaction(token_contract, "approve", [self.arb_contract.address, MAX_UINT256], 0)
+                tx = self._create_transaction(token_contract, "approve", [self.arb_contract.address, MAX_UINT256], 0, self.cfg.SUPPORTS_EIP1559)
                 self._update_transaction(tx)
                 raw_tx = self._sign_transaction(tx)
                 tx_hash = self._send_regular_transaction(raw_tx)
                 self._wait_for_transaction_receipt(tx_hash)
 
-    def _create_transaction(self, contract, fn_name: str, args: list, value: int) -> dict:
-        return {
-            "type": 2,
-            "value": value,
-            "chainId": self.chain_id,
-            "from": self.wallet_address,
-            "to": contract.address,
-            "data": contract.encode_abi(fn_name=fn_name, args=args),
-            "nonce": self.cfg.w3.eth.get_transaction_count(self.wallet_address)
-        }
+    def _create_transaction(self, contract, fn_name: str, args: list, value: int, supports_eip1559: bool) -> dict:
+        if supports_eip1559:
+            return {
+                "type": 2,
+                "value": value,
+                "chainId": self.chain_id,
+                "from": self.wallet_address,
+                "to": contract.address,
+                "data": contract.encode_abi(fn_name=fn_name, args=args),
+                "nonce": self.cfg.w3.eth.get_transaction_count(self.wallet_address)
+            }
+        else:
+            return {
+                "value": value,
```

Comment:
Missing `type` (0 or 1, need to find out with Telos which one is supported)

Reply:
Actually not.
When passing type as 1: ValueError: {'message': 'Invalid Transaction: Calling from_big_endian with oversized array', 'code': 3}
When passing type as 0: TypeError: Unknown Transaction type: 0
When passing nothing as I have it, the arb executes immediately